hack/shellcheck-test.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+  
+# Check for shell syntax & style.
+
+test_syntax() {
+        bash -n "${1}"
+}
+test_shellcheck() {
+        if [[ "${SHELLCHECK}" ]]; then
+                # only look for the "flag" on comment lines
+                # without this we can match our own code :-)
+                if grep -q '^#.*OCS-OPERATOR-SKIP-SHELLCHECK' "${1}"; then
+                        return 0
+                fi
+                shellcheck -x -e SC2181,SC2029,SC1091,SC1090,SC2012 "${1}"
+        else
+                return 0
+        fi
+}
+
+SHELLCHECK="$(command -v shellcheck 2>/dev/null)"
+
+SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+
+BASE_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+if [[ -z "${SHELLCHECK}" ]]; then
+        echo "warning: could not find shellcheck ... installing shellcheck" >&2
+        scversion="stable" # or "v0.4.7", or "latest"
```

Comment:
Nit, but as long as we're cleaning up other concerns, the comment here could be misleading. It could read to someone who doesn't know shellcheck as "we're pulling version stable, which is the same as v0.4.7". It's unlikely to cause confusion, but it could.

It's fine to leave the comment in: we don't likely need it, but it might be useful to someone. Still, the shellcheck repo has the comment "Or v0.4.7 for that version, or latest for daily", which is a bit clearer.